File: weather-feeder/src/store/event_store_reader.py
import os


class EventStoreReader:
    def __init__(self, event_store_path: str):
        self.event_store_path = event_store_path

    def load(self, topic: str, handler) -> None:
        topic_path = os.path.join(self.event_store_path, topic)
        if not os.path.exists(topic_path):
            print(f"[event-store-reader] Path not found: {topic_path}")
            return

        event_files = []
        for root, _, files in os.walk(topic_path):
            for file in files:
                if file.endswith('.events'):
                    event_files.append(os.path.join(root, file))

        for file_path in sorted(event_files):
            self._load_file(file_path, handler)

    def _load_file(self, file_path: str, handler) -> None:
        try:
            with open(file_path, 'r') as f:
                for line in f:
                    line = line.strip()
                    if line:
                        handler(line)
        except Exception as e:
            print(f"[event-store-reader] Failed to read file {file_path}: {e}")

File: business-unit/src/main/java/es/ulpgc/dacd/business/controller/EventStoreReader.java
package es.ulpgc.dacd.business.controller;

import es.ulpgc.dacd.business.handler.EventHandler;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.logging.Logger;

public class EventStoreReader {

    private static final Logger logger = Logger.getLogger(EventStoreReader.class.getName());

    private final String eventStorePath;

    public EventStoreReader(String eventStorePath) {
        this.eventStorePath = eventStorePath;
    }

    public void load(String topic, EventHandler handler) {
        Path topicPath = Path.of(eventStorePath, topic);
        if (!Files.exists(topicPath)) return;
        try {
            Files.walk(topicPath)
                    .filter(p -> p.toString().endsWith(".events"))
                    .sorted()
                    .forEach(file -> loadFile(file, handler));
        } catch (IOException e) {
            logger.severe("Failed to read event store: " + e.getMessage());
        }
    }

    private void loadFile(Path file, EventHandler handler) {
        try (BufferedReader reader = new BufferedReader(new FileReader(file.toFile()))) {
            String line;
            while ((line = reader.readLine()) != null) {
                if (!line.isBlank()) handler.handle(line);
            }
        } catch (IOException e) {
            logger.severe("Failed to read file " + file + ": " + e.getMessage());
        }
    }
}
